beep/protocol/maccor_to_arbin.py

```diff
@@ -40,26 +45,51 @@ def create_sdu(self, sdu_input_name, sdu_output_name):
             sdu_input_name (str): the full path of the schedule file to use as a
                 shell for the steps
             sdu_output_name (str): the full path of the schedule file to output
+            global_v_range (list): Global safety range for voltage in volts [min, max]
+            global_temp_range (list): Global safety range for temperature in Celcius [min, max]
+            global_current_range (list): Global safety range for current in Amps [min, max]
 
         """
         schedule = Schedule.from_file(sdu_input_name)
         # sdu_dict = Schedule.from_file(sdu_input_name)
 
-        keys = list(schedule["Schedule"].keys())
+        keys = list(schedule['Schedule'].keys())
+        safety_data = []
+        safety_key = []
         for key in keys:
-            if "Schedule" in key:
-                del schedule["Schedule"][key]
+            if bool(re.match('Step[0-9]+', key)):
+                del schedule['Schedule'][key]
+            if bool(re.match('UserDefineSafety[0-9]+', key)):
+                safety_data.append(deepcopy(schedule['Schedule'][key]))
+                safety_key.append(key)
+                del schedule['Schedule'][key]
```

Comment:
Can this be turned into an if...else if? It will save the tiniest bit of time by not evaluating the second if we matched the first. Also, if we expect to match one more than the other, it should go first.

Reply:
Yes, that's a good idea, thanks!